lib/pipelines/code-pipeline.ts

```diff
@@ -51,12 +51,45 @@ export interface CodeCommitSourceRepository
     codeCommitOptions?: cdkpipelines.CodeCommitSourceOptions;
 }
 
-export function isCodeCommitRepo(repo: GitHubSourceRepository | CodeCommitSourceRepository): boolean{
-    if (Object.prototype.hasOwnProperty.call(repo, "codeCommitRepoName")) {
-        return true;
-    } else {
-        return false;
-    }
+export interface CodeStarConnectionRepository
+  extends Omit<
+    ApplicationRepository,
+    'credentialsType' | 'credentialsSecretName '
+  > {
+  /**
+   * The ARN of the CodeStar Connection.
+   */
+  codeStarConnectionArn: string;
+  /**
+   * The owner of the repository for the pipeline
+   */
+  owner?: string;
+}
+
+export function isCodeCommitRepo(
+  repo:
+    | GitHubSourceRepository
+    | CodeCommitSourceRepository
+    | CodeStarConnectionRepository
+): boolean {
+  if (Object.prototype.hasOwnProperty.call(repo, 'codeCommitRepoName')) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+export function isCodeStarConnection(
+  repo:
+    | GitHubSourceRepository
+    | CodeCommitSourceRepository
+    | CodeStarConnectionRepository
+): boolean {
+  if (Object.prototype.hasOwnProperty.call(repo, 'codeStarConnectionArn')) {
```

Comment:
Minor: instead of having this if/else statement why not just do 

```
return Object.prototype.hasOwnProperty.call(repo, 'codeStarConnectionArn');
```